experimental/QuadFormAndIsom/src/embeddings.jl

```diff
@@ -0,0 +1,1325 @@
+  else
+    sors = _subgroups_orbit_representatives_and_stabilizers(Vinq, O, order(H), f)
+  end
+  return filter(d -> is_isometric_with_isometry(domain(d[1]), H)[1], sors)
+end
+
+#################################################################################
+#
+# Primitive embeddings for primary lattices
+#
+#################################################################################
+
+# We compute representatives of isomorphism classes of primitive extensions
+# M \oplus N \to L, where we glue along HM and HN which are respectively
+# isometric and anti-isometric to H.
+#
+# We follow the second definition of Nikulin, i.e. we classify up to the action
+# of O(N). If `classification == :sublat`, we also classify them up to the
+# action of O(M). If `classification == :first`, we return the first embedding
+# computed.
+function _isomorphism_classes_primitive_extensions(N::ZZLat, M::ZZLat, H::TorQuadModule, classification::Symbol)
+  @hassert :ZZLatWithIsom 1 classification in [:first, :emb, :sublat]
+  results = Tuple{ZZLat, ZZLat, ZZLat}[]
+  prim, p = is_primary_with_prime(H)
+  el = prim ? is_elementary(H, p) : false
+
+  qN = discriminant_group(N)
+  GN, _ = image_in_Oq(N)
+
+  qM = discriminant_group(M)
+  if classification == :emb
+    GM = Oscar._orthogonal_group(qM, ZZMatrix[matrix(id_hom(qM))])
+  else
+    GM, _ = image_in_Oq(M)
+  end
+
+  D, inj = direct_sum(qN, qM)
+  qNinD, qMinD = inj
+  OD = orthogonal_group(D)
+
+  if el
+    VN, VNinqN = _get_V(id_hom(qN), minimal_polynomial(identity_matrix(QQ,1)), p)
+    VM, VMinqM = _get_V(id_hom(qM), minimal_polynomial(identity_matrix(QQ, 1)), p)
+  elseif prim
+    VN, VNinqN = primary_part(qN, p)
+    VM, VMinqM = primary_part(qM, p)
+  else
+    VN, VNinqN = qN, id_hom(qN)
+    VM, VMinqM = qM, id_hom(qM)
+  end
+
+  subsN = _classes_automorphic_subgroups(VNinqN, GN, rescale(H, -1))
+  @hassert :ZZLatWithIsom 1 !isempty(subsN)
+  subsM = _classes_automorphic_subgroups(VMinqM, GM, H)
+  @hassert :ZZLatWithIsom 1 !isempty(subsM)
+
+  for H1 in subsN, H2 in subsM
+    ok, phi = is_anti_isometric_with_anti_isometry(domain(H1[1]), domain(H2[1]))
+    @hassert :ZZLatWithIsom 1 ok
+
+    HNinqN, stabN = H1
+    HNinD = compose(HNinqN, qNinD)
+    HN = domain(HNinqN)
+    OHN = orthogonal_group(HN)
+
+    HMinqM, stabM = H2
+    HMinD = compose(HMinqM, qMinD)
+    HM = domain(HMinqM)
+    OHM = orthogonal_group(HM)
+
+    actN = hom(stabN, OHN, [OHN(restrict_automorphism(x, HNinqN)) for x in gens(stabN)])
+    actM = hom(stabM, OHM, [OHM(restrict_automorphism(x, HMinqM)) for x in gens(stabM)])
+    imM, _ = image(actM)
+
+    stabNphi = AutomorphismGroupElem{TorQuadModule}[OHM(compose(inv(phi), compose(hom(actN(g)), phi))) for g in gens(stabN)]
+    stabNphi, _ = sub(OHM, stabNphi)
+
+    reps = double_cosets(OHM, stabNphi, imM)
+    @vprint :ZZLatWithIsom 1 "$(length(reps)) isomorphism classe(s) of primitive extensions\n"
```

Comment:
```suggestion
    @vprintln :ZZLatWithIsom 1 "$(length(reps)) isomorphism classe(s) of primitive extensions"
```
Here and at other places